docs/table/construct_table.rst

```diff
@@ -1303,15 +1297,7 @@ breakage in this case. ::
           cols = list(self.values())
           names = list(self.keys())
 
-          # If returning a reference to existing data (copy=False) and
-          # strict_copy=True, make sure that each column is a numpy ndarray.
-          # If a column is a Python list or tuple then it must be copied for
-          # representation in an astropy Table.
-
-          if not copy and strict_copy:
-              for name, col in zip(names, cols):
-                  if not isinstance(col, np.ndarray):
-                      raise ValueError(f'cannot have copy=False because column {name} is '
-                                       'not an ndarray')
+          if mask_invalid:
+              cols = [Masked(col, mask=np.isnan(col)) for col in cols]
```

Comment:
Even though this is a toy example, it seems like you should only make cols `Masked` if need be:
```suggestion
              cols = [
                  Masked(col, mask=mask) if np.any(mask := np.isnan(col)) else col 
                  for col in cols
              ]
```

Reply:
good point, and love that walrus !